lib/Frontend/Frontend.cpp

```diff
@@ -209,30 +209,35 @@ SerializationOptions CompilerInvocation::computeSerializationOptions(
     serializationOpts.ExtraClangOptions = getClangImporterOptions().ExtraArgs;
   }
 
-  // '-plugin-path' options.
-  for (const auto &path : getSearchPathOptions().PluginSearchPaths) {
-    serializationOpts.PluginSearchPaths.push_back(path);
-  }
-  // '-external-plugin-path' options.
-  for (const ExternalPluginSearchPathAndServerPath &pair :
-       getSearchPathOptions().ExternalPluginSearchPaths) {
-    serializationOpts.ExternalPluginSearchPaths.push_back(
-        pair.SearchPath + "#" +
-        pair.ServerPath);
-  }
-  // '-load-plugin-library' options.
-  for (const auto &path :
-       getSearchPathOptions().getCompilerPluginLibraryPaths()) {
-    serializationOpts.CompilerPluginLibraryPaths.push_back(path);
-  }
-  // '-load-plugin-executable' options.
-  for (const PluginExecutablePathAndModuleNames &pair :
-       getSearchPathOptions().getCompilerPluginExecutablePaths()) {
-    std::string optStr = pair.ExecutablePath + "#";
-    llvm::interleave(
-        pair.ModuleNames, [&](auto &name) { optStr += name; },
-        [&]() { optStr += ","; });
-    serializationOpts.CompilerPluginExecutablePaths.push_back(optStr);
+  // FIXME: Preserve the order of these options.
```

Comment:
Are you intentionally trying to avoid changing the module serialization format here?

Reply:
Yes, my intension is doing it in a followup.